common/src/main/java/io/netty/util/AbstractReferenceCounted.java

```diff
@@ -51,27 +53,15 @@ protected final void setRefCnt(int refCnt) {
 
     @Override
     public ReferenceCounted retain() {
-        for (;;) {
-            int refCnt = this.refCnt;
-            final int nextCnt = refCnt + 1;
-
-            // Ensure we not resurrect (which means the refCnt was 0) and also that we encountered an overflow.
-            if (nextCnt <= 1) {
-                throw new IllegalReferenceCountException(refCnt, 1);
-            }
-            if (refCntUpdater.compareAndSet(this, refCnt, nextCnt)) {
-                break;
-            }
-        }
-        return this;
+        return retain0(1);
     }
 
     @Override
     public ReferenceCounted retain(int increment) {
-        if (increment <= 0) {
-            throw new IllegalArgumentException("increment: " + increment + " (expected: > 0)");
-        }
+        return retain0(checkPositive(increment, "increment"));
+    }
 
+    private ReferenceCounted retain0(int increment) {
         for (;;) {
```

Comment:
@Scottmitch this utility stuff feels ugly. let us just duplicate the code for now.